docs/store.js

```diff
@@ -302,16 +299,24 @@ var firebaseCalendarStorage = function(params) {
         
         self.startWork();
         var addWriterP = self.addWriter(self.user, calendarObj)
-        .then(function() {
-            self.addReader(self.user, calendarObj);
-            self.setActiveById(calendarObj.state.uniqueId);
-            self.save(calendarObj);
-            self.endWork();
-        })
-        .catch(function(err) {
-            console.error("Problems initializing calendar " + err);
-            self.endWork();
-        });
+            .then(function() {
+                var addReaderP = self.addReader(self.user, calendarObj);
+                var setActiveP = self.setActiveById(calendarObj.state.uniqueId);
+                var saveP = self.save(calendarObj);
+                Promise.all([addReaderP, setActiveP, saveP])
+                .then(function() {
```

Comment:
I find that lining up the methods vertically if they are going to operate on the same object is an easy way to organize it.
```
Promise.all([...])
       .then(function() {
          ...
       })
       .catch(function() {});
```
However you decide to organize it, it should be such that lines that begin in the column indicate new *statements*, not expressions. If an expression continues onto a second line, it should be indented somehow to give a visual indication that this is all a single computation that will return a single value.